### DISCUSSION TITLE
Chezmoi config template prompting multiple times using promptBoolOnce

### DISCUSSION THREAD

**Discussion by dslatkin** [twpayne/chezmoi#2373] [Q&A] [answered]
Hi all, just posting here for some help. Still pretty new to chezmoi, totally new to Go and `text/template`, so wondering if I'm doing something wrong. I'm calling `promptStringOnce` in my config template which behaves as expected, but `promptBoolOnce` which is otherwise the same seems to not be working properly...

[This is what my chezmoi config temaplate looks like.](https://github.com/dslatkin/dotfiles/blob/4de2f0fb7fed97ffa6e5c7873fa9dfedb9f49f8d/.chezmoi.toml.tmpl)

When I either:

1. `chezmoi init` on a new machine
2. `rm ~/.config/chezmoi/chezmoi.topml` then `chezmoi init`

...I'm prompted for the following:

```
Name for git commits? ...
Include personal settings? ...
Include work settings? ...
Include AI tooling? ...
```

...but if I run it again, I'm not prompted for name (string) and only prompted for the others (bools):

```
Include personal settings? ...
Include work settings? ...
Include AI tooling? ...
```

Since my config file is almost otherwise identical, my first guess is this might be a bug? Or am I doing something wrong? Thanks!

**Comment by twpayne** [comment, marked as answer]
I think you're hitting #463, i.e. the keys in `data` are being converted to lowercase. `name` happens to work because its already all lowercase, whereas the other keys use a mix of cases.

Running `chezmoi data` will show you what data chezmoi is actually reading from the system and your config file.

**Reply by dslatkin** [reply]
That's exactly it, thank you! Maybe checking case differences for data with `chezmoi doctor` would make sense until the final fixed it released? I would have noticed that before starting this discussion. If you think it would be a pretty straightforward contribution, I wouldn't mind giving it a go. 🙂

**Reply by twpayne** [reply]
Thank you very much for the offer! Checking for case differences is not 100% reliable as it depends how the template data is used. From memory, spf13/viper only converts top-level keys to lowercase, not all keys, and so mixed-case keys can be OK in some circumstances, so a reliable doctor check might be difficult to write.

I think the real fix here is for chezmoi to fix remove the dependency on on spf13/viper and read its configuration file itself, preserving case. I believe (but am not sure) that this might also fix #2046.

**Reply by twpayne** [reply]
Fix in progress in #2376.

**Reply by dslatkin** [reply]
Awesome, thank you for all your hard work on this project! Loving chezmoi. 😁
